File: frontend/lib/models/owner_profile_dto.dart
class OwnerProfileDto {
  final String inn;
  final String address;
  final String? legalName;
  final String? contactPerson;
  final String? contactPhone;
  final String? contactEmail;

  OwnerProfileDto({
    required this.inn,
    required this.address,
    this.legalName,
    this.contactPerson,
    this.contactPhone,
    this.contactEmail,
  });

  Map<String, dynamic> toJson() => {
        'inn': inn,
        'address': address,
        'legalName': legalName,
        'contactPerson': contactPerson,
        'contactPhone': contactPhone,
        'contactEmail': contactEmail,
      };

  factory OwnerProfileDto.fromJson(Map<String, dynamic> json) {
    return OwnerProfileDto(
      inn: json['inn'] as String,
      address: json['address'] as String? ?? '',
      legalName: json['legalName'] as String?,
      contactPerson: json['contactPerson'] as String?,
      contactPhone: json['contactPhone'] as String?,
      contactEmail: json['contactEmail'] as String?,
    );
  }
}

File: backend/src/main/java/ru/bowling/bowlingapp/DTO/OwnerProfileDTO.java
package ru.bowling.bowlingapp.DTO;

import jakarta.validation.constraints.*;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OwnerProfileDTO {
    @NotBlank
    private String inn;
    
    private String legalName;
    private String contactPerson;
    private String contactPhone;
    private String contactEmail;
}

